packages/node/src/integrations/http.ts

```diff
@@ -145,21 +147,70 @@ function addRequestBreadcrumb(event: string, url: string, req: http.IncomingMess
 }
 
 /**
- * Function that can combine together a url that'll be used for our breadcrumbs.
+ * Strip the query string and fragment off of a given URL or path (if present)
  *
- * @param options url that should be returned or an object containing it's parts.
- * @returns constructed url
+ * @param urlPath Path including possible query string and/or fragment
+ * @returns Path without query string or fragment
  */
-function extractUrl(options: string | http.ClientRequestArgs): string {
-  if (typeof options === 'string') {
-    return options;
+export function stripUrlPath(urlPath: string): string {
+  if (urlPath.includes('?') || urlPath.includes('#')) {
+    // in theory, the fragment should always come after the query string, but you never know, so make the # a ? so the
+    // split will always happen if either is there
+    return urlPath.replace('#', '?').split('?')[0];
```

Comment:
No need to do string replacements here.

> 3.3.  Path
> 
>    The path component contains data, usually organized in hierarchical
>    form, that, along with data in the non-hierarchical query component
>    (Section 3.4), serves to identify a resource within the scope of the
>    URI's scheme and naming authority (if any).  The path is terminated
>    by the first question mark ("?") or number sign ("#") character, or
>    by the end of the URI.

https://www.ietf.org/rfc/rfc3986.txt

Reply:
If we want to match on both anyway, you can write this whole function as
```js
return urlPath.split(/[\?#]/, 1)[0];
```
It'll return original input if nothing will match anyway, so `includes` is redundant.

Reply:
`indexOf` and slicing would be more efficient.

In general, if we're only interested in the first part of the split, then we should use the `limit` argument https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split.

Reply:
That's what I used (limit) :)

Reply:
Ah - TIL that you can split on a regex! (And about `limit`.)

According to the MDN docs, I think we don't even need to index into it:

> If limit is 1, str.split(separator)[0] is returned

`return urlPath.split(/[\?#]/, 1);` seems like it should do it.

Reply:
(Yeah - that just seems to be wrong. Tested in all three browsers and node. Filed an issue: https://github.com/mdn/sprints/issues/3648)